quickwit-search/src/root.rs

```diff
@@ -709,7 +718,12 @@ mod tests {
             .expect_leaf_search()
             .times(2)
             .returning(|leaf_search_req: quickwit_proto::LeafSearchRequest| {
-                if leaf_search_req.split_ids == vec!["split1".to_string()] {
+                let split_ids = leaf_search_req
+                    .split_metadata
+                    .iter()
+                    .map(|metadata| metadata.split_id.to_string())
+                    .collect_vec();
+                if split_ids == vec!["split1".to_string()] {
```

Comment:
Vec on the left hand is useless here.

```&split_ids == &["split1".to_string()]```

If we keep doing it in test, we'll end up doing it in code.